microsoft/testsuites/core/timesync.py

```diff
@@ -64,3 +68,45 @@ def timesync_validate_ptp(self, environment: Environment, node: Node) -> None:
                 assert_that(chrony_results.stdout).described_as(
                     "Chrony config file should use the symlink /dev/ptp_hyperv."
                 ).contains(self.hyperv_ptp_udev_rule)
+
+    @TestCaseMetadata(
+        description="""
+        This test is to check -
+            1. Check clock source name shown as hyperv_clocksource_tsc_page, except:
+                1.1 If it is centos/rhel/oracle below 7.5 + with LIS driver, it
+                    shown as lis_hv_clocksource_tsc_page.
+                1.2 If it is centos/rhel/oracle below 7.2 + without LIS driver, it
+                    shown as hyperv_clocksource.
+            2. Check CPU flag contains tsc from /proc/cpuinfo.
+            3. Check clocksource name shown up in dmesg.
+        """,
+        priority=2,
+    )
+    def timesync_check_timesource(self, environment: Environment, node: Node) -> None:
+        # 1. Check clock source name shown as hyperv_clocksource_tsc_page
+        clocksource = "hyperv_clocksource_tsc_page"
+        cat = node.tools[Cat]
+        clock_source_result = cat.run(self.current_clocksource)
+        os_version = node.os.os_version
+        os_verion_convert = (
+            os_version.release.split(".")[0] + "." + os_version.release.split(".")[1]
+        )
+        if isinstance(node.os, Redhat):
+            if node.os._package_exists("kmod-microsoft"):
+                if float(os_verion_convert) < 7.5:
```

Comment:
you should be able to directly compare it like 
if float(self.node.os.os_version.release) < 7.5:

Reply:
@sharsonia  Can we provide a method in os to compare the os_version? For most versions, they can be compared like float, some of them may need be customized, for example "12.3.5". But it looks doable.

Reply:
yeah, I used float(self.node.os.os_version.release), but failed when the release returns 8.3.2011